File: cbs-server/src/main/java/com/evil/cbs/domain/Movie.java
package com.evil.cbs.domain;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.time.Duration;
import java.util.HashSet;
import java.util.Set;


@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "movies", schema = "cbs")
@AllArgsConstructor(access = AccessLevel.PACKAGE)
@NoArgsConstructor(access = AccessLevel.PACKAGE)
@Builder
public class Movie extends AbstractEntity {
    @Column(name = "name")
    private String name;
    @Column(name = "description")
    private String description;
    @Column(name = "imdb_rating")
    private String imdbRating;
    @Column(name = "movie_duration")
    private Duration movieDuration;
    @ElementCollection
    private Set<String> directors = new HashSet<>();
    @ElementCollection
    private Set<String> actors = new HashSet<>();
    @Column(name = "image_path", nullable = true)
    private String imagePath;

}

File: cbs-client/src/app/model/movie.ts
import {MovieSession} from "./movie-session";

export class Movie {
  id: number
  name: string;
  description: string;
  imdbRating: string;
  movieDuration: number;
  movieSession: MovieSession;
  directors: Array<string>;
  actors: Array<string>;
  imagePath: string;


  constructor(id: number, name: string, description: string, imdbRating: string, movieDuration: number, movieSession: MovieSession, directors: Array<string>, actors: Array<string>, imagePath: string) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.imdbRating = imdbRating;
    this.movieDuration = movieDuration;
    this.movieSession = movieSession;
    this.directors = directors;
    this.actors = actors;
    this.imagePath = imagePath;
  }
}
